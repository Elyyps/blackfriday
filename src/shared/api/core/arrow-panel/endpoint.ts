import { generateArrowPanelData } from "./generate-dummy-data";

export const generateTabsDummyData = generateArrowPanelData();
