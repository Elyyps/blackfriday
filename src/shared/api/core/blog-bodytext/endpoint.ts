import { generateDummyBlogBodyText } from "./generate-dummy-data";

export const generateBlogBodyTextData = generateDummyBlogBodyText;
