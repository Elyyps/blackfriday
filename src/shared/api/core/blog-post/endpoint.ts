import { generateDummyBlogPost, generate2DummyBlogPosts } from "./generate-dummy-data";

export const generateBlogPostData = generateDummyBlogPost;
export const generate2BlogPostsData = generate2DummyBlogPosts;
