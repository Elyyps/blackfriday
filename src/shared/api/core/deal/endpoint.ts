import { Deal } from "./deal";
import { generateDemoDealDummyData } from "./generate-dummy-data";
import { FilterItem } from "../filter/filter-item";
import { shuffle } from "@app/util/array";
/* tslint:disable */
export interface IDealResult {
  deals: Deal[];
  totalResults: number;
}

const initialStoresResult = [...generateDemoDealDummyData()];

const getStores = (
  skip: number,
  take: number,
  storeFilters: FilterItem[],
  categoryFilters: FilterItem[],
  brandFilters: FilterItem[],
  sortBy?: string
): IDealResult => {
  const selectedCategoryFilters = getSelectedFilters(categoryFilters);
  const selectedBrandFilters = getSelectedFilters(brandFilters);
  const selectedStoreFilters = getSelectedFilters(storeFilters);

  let result = [...initialStoresResult];

  if (selectedCategoryFilters !== undefined) {
    result = result.filter(deal => {
      const items = deal.categories.filter(category => {
        return selectedCategoryFilters.includes(category);
      });

      return items.length > 0;
    });
  }

  if (selectedBrandFilters !== undefined) {
    result = result.filter(deal => {
      const items = deal.availableBrands.filter(category => {
        return selectedBrandFilters.includes(category);
      });

      return items.length > 0;
    });
  }

  if (selectedStoreFilters !== undefined) {
    result = result.filter(deal => {
      const items = deal.availableBrands.filter(category => {
        return selectedStoreFilters.includes(category);
      });

      return items.length > 0;
    });
  }

  const totalResults = result.length;

  result = result.splice(skip, take);

  switch (sortBy) {
    case "Relevantie":
      // Keep default sorting
      break;
    case "Nieuwste":
      result.reverse();
      break;
    case "Populair":
      result = shuffle(result);
      break;
    case "Winkels A - Z":
      result = result.sort((a, b) => {
        if (a.name < b.name) {
          return -1;
        }
        if (b.name < a.name) {
          return 1;
        }

        return 0;
      });
      break;
    case "Winkels Z - A":
      result = result.sort((a, b) => {
        if (a.name > b.name) {
          return -1;
        }
        if (b.name > a.name) {
          return 1;
        }

        return 0;
      });
      break;
    default:
      /* tslint:disable:switch-final-break  */
      // keep default sorting
      break;
  }

  return {
    deals: result,
    totalResults
  };
};

const getSelectedFilters = (filterItems: FilterItem[]): string[] | undefined => {
  const selectedFilterItems = filterItems.filter(item => item.isSelected);

  if (selectedFilterItems && selectedFilterItems.length > 0) {
    return selectedFilterItems.map(item => item.displayName);
  }

  return undefined;
};

const getAmountForCategory = (category: string): number =>
  initialStoresResult.filter(deal => deal.categories.indexOf(category) > -1).length;

const getAmountForBrand = (brand: string): number =>
  initialStoresResult.filter(deal => deal.availableBrands.indexOf(brand) > -1).length;

const getAmountForStore = (store: string): number =>
  initialStoresResult.filter(deal => deal.stores.indexOf(store) > -1).length;

export const DealApi = {
  getStores,
  initialStoresResult,
  getAmountForCategory,
  getAmountForBrand,
  getAmountForStore
};
