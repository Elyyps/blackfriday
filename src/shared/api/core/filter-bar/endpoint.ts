import { FilterBar } from "./filter-bar";
import { generateDummyFilterBar, generateEmptyDummyFilterBar } from "./generate-dummy-data";

export const generateFilterBarData = generateDummyFilterBar;
export const generateEmptyFilterBarData = generateEmptyDummyFilterBar;
