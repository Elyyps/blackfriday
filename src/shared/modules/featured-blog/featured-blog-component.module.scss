.featured-blog {
  &__card {
    width: 48%;
    @media (max-width: $breakpoint-small) {
      width: 100%;
      margin-bottom: 20px;
    }
  }

  &__title {
    margin-bottom: 36px;
    @media (max-width: $breakpoint-small) {
      margin-bottom: 16px;
    }
  }

  &__body {
    width: 100%;
    display: flex;
    justify-content: space-between;
    flex-direction: row;
    @media (max-width: $breakpoint-small) {
      flex-direction: column;
    }
  }
}
