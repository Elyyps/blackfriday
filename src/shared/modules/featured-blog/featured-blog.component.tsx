import * as React from "react";
import styles from "./featured-blog-component.module.scss";
import { FeaturedBlogModule } from "@app/api/modules/featured-blog/featured-blog";
import { FeaturedBlogItemsComponent } from "@app/core/featured-blog-items";

export interface IFeaturedBlogComponentProps {
  featuredBlog: FeaturedBlogModule;
}

const FeaturedBlogComponent = (props: IFeaturedBlogComponentProps) => (
  <div className={styles["featured-blog"]}>
    <div className="uk-container">
      {props.featuredBlog.title && <h2 className={styles["featured-blog__title"]}>{props.featuredBlog.title}</h2>}
      <div className={styles["featured-blog__body"]}>
        {props.featuredBlog.items &&
          props.featuredBlog.items.map((item, key) => (
            <div key={key}>
              <FeaturedBlogItemsComponent item={item} />
            </div>
          ))}
      </div>
    </div>
  </div>
);

export { FeaturedBlogComponent };
