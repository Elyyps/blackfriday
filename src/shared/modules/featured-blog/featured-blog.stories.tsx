import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { FeaturedBlogComponent } from "./featured-blog.component";
import { generateFeaturedBlogData } from "@app/api/modules/featured-blog/endpoint";
const data = generateFeaturedBlogData();
storiesOf("FeaturedBlog", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => <FeaturedBlogComponent featuredBlog={data} />);
