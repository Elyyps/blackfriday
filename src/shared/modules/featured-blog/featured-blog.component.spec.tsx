import React from "react";
import { shallow } from "enzyme";
import { FeaturedBlogComponent } from "./featured-blog.component";
import { generateFeaturedBlogData } from "@app/api/modules/featured-blog/endpoint";
const data = generateFeaturedBlogData();
describe("[FeaturedBlog]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<FeaturedBlogComponent featuredBlog={data} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
