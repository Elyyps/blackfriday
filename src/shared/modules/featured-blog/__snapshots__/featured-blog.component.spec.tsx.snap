// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[FeaturedBlog] should render component without crashing 1`] = `
<div
  className="featured-blog"
>
  <div
    className="uk-container"
  >
    <h2
      className="featured-blog__title"
    >
      Ook interessant om te lezen
    </h2>
    <div
      className="featured-blog__body"
    >
      <div
        className="featured-blog__card"
        key="0"
      >
        <LinkComponent
          to="/blog-single"
        >
          <div
            className="featured-blog__card-image"
          >
            <ImageComponent
              src="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/backFridayAd.png"
            />
          </div>
          <h3
            className="featured-blog__card-title"
          >
            Vijf voorspellingen van Black Friday Nederland
          </h3>
        </LinkComponent>
      </div>
      <div
        className="featured-blog__card"
        key="1"
      >
        <LinkComponent
          to="/blog-single"
        >
          <div
            className="featured-blog__card-image"
          >
            <ImageComponent
              src="https://viper-development-images.s3-eu-west-1.amazonaws.com/blackFriday/backFridayAd.png"
            />
          </div>
          <h3
            className="featured-blog__card-title"
          >
            Top Black Friday deals van vorig jaar
          </h3>
        </LinkComponent>
      </div>
    </div>
  </div>
</div>
`;
